src/qextserialenumerator_osx.cpp

```diff
@@ -114,25 +116,29 @@ bool QextSerialEnumeratorPrivate::getServiceDetailsOSX(io_object_t service, Qext
     // vendor/product IDs and the product name, if available
     io_registry_entry_t parent;
     kern_return_t kernResult = IORegistryEntryGetParentEntry(service, kIOServicePlane, &parent);
-    while (kernResult == KERN_SUCCESS && !vendorIdAsCFNumber && !productIdAsCFNumber && !serialNumberAsCFString) {
+    while (kernResult == KERN_SUCCESS && (!vendorIdAsCFNumber || !productIdAsCFNumber || !serialNumberAsCFString)) {
```

Comment:
I feel like this could be tweaked to read !(A && B && C) which feels clearer to me somehow.